flight/Modules/Attitude/attitude.c

```diff
@@ -1208,6 +1210,32 @@ static int32_t updateAttitudeINSGPS(bool first_run, bool outdoor_mode)
 		gps_vel_updated = false;
 	}
 
+	// If either vel or pos is updated, update the variances.
+	if (gps_updated || gps_vel_updated) {
+		// Typical good pos 'accuracy' values are 2.5-3.5.  Early
+		// flight is near 4.0.
+		// Typical bad pos 'accuracy' values are 10+
+		// The values here are fudged.  Basically, when GPS is
+		// "good" we'll have a lower variance than the nominal
+		// setting.  But from there it will increase really quickly.
+		// sqrt(.5) / 3.5 =~ 0.181f 
+		float pos_var = insSettings.GpsVar[INSSETTINGS_GPSVAR_POS] * 
+			(0.6f + powf(gpsData.Accuracy * 0.180f, 2));
+
+		// A good value of speed accuracy in flight is 0.35-0.5. 
+		// sqrt(.5) / .5 =~ 1.414
+		float speed_var = insSettings.GpsVar[INSSETTINGS_GPSVAR_VEL] *
+			(0.5f + powf(gpsVelData.Accuracy * 1.414f, 2));
```

Comment:
so e.g. here we can make it:
speed_var = insSettings.GpsVarMin + insSettings.GpsVarScale * powf(gpsVelData.Accuracy, 2)

set GpsVarMin to 0.0005 and GpsVarScale to 0.002

that produces the identical mapping from accuracy to speed_var but has a bit more flexibility if we want it later